server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -205,7 +206,7 @@ public IndexService(
                 this.indexSortSupplier = () -> indexSettings.getIndexSortConfig()
                     .buildIndexSort(
                         mapperService::fieldType,
-                        (fieldType, searchLookup) -> indexFieldData.getForField(fieldType, indexFieldData.index().getName(), searchLookup)
+                        (fieldType, searchLookup) -> indexFieldData.getForField(fieldType, FieldDataContext.noRuntimeFields("index sort"))
```

Comment:
IndexSortConfig throws an exception too by providing a supplier of search lookup that throws exception. Which exception comes first and do we need both?